packages/hub/node-tests/cards-test.js

```diff
@@ -0,0 +1,247 @@
+      });
+
+      it(`allows for a content-type's field to specify a related-types that refer to a content-type within the card schema`, async function () {
+        let factory = new JSONAPIFactory();
+        factory.addResource('data-sources').withAttributes({ 'source-type': 'person-card' });
+        factory.addResource('data-sources').withAttributes({ 'source-type': 'article-card' });
+
+        env = await createDefaultEnvironment(`${__dirname}/../../../tests/stub-cards/valid-cards`, factory.getModels());
+        searchers = env.lookup('hub:searchers');
+
+        let schema = await searchers.get(env.session, 'local-hub', 'fields', 'local-hub::article-card::categories');
+        expect(schema.data.relationships['related-types'].data).to.eql([
+          { type: 'content-types', id: 'local-hub::article-card::categories'
+         }
+        ]);
+      });
+    });
+
+    describe('invalid schema scenarios', function () {
+      // no need to teardown() since env is never created in these scenarios
+
+      it('throws when card schema create non-schema models', async function () {
+        let factory = new JSONAPIFactory();
+        factory.addResource('data-sources').withAttributes({ 'source-type': 'person-card' });
+
+        let error;
+        try {
+          await createDefaultEnvironment(`${__dirname}/../../../tests/stub-cards/non-schema-models-in-schema-file`, factory.getModels());
+        } catch (e) {
+          error = e;
+        }
+
+        expect(error.message).to.match(/Non-schema models are not allowed in schemas/);
+      });
+
+      it('throws when card schema models are not scoped to the source and package that they originate from (which can happen when JSONAPIFactory is not given source and package context)', async function () {
+        let factory = new JSONAPIFactory();
+        factory.addResource('data-sources').withAttributes({ 'source-type': 'person-card' });
+
+        let error;
+        try {
+          await createDefaultEnvironment(`${__dirname}/../../../tests/stub-cards/jsonapi-factory-missing-card-context`, factory.getModels());
+        } catch (e) {
+          error = e;
+        }
+
+        expect(error.message).to.match(/has schema models that are not scoped to this data source and package/);
+      });
+
+      it('throws when card-defintions id does not match source::package of the schema file name', async function () {
+        let factory = new JSONAPIFactory();
+        factory.addResource('data-sources').withAttributes({ 'source-type': 'person-card' });
+
+        let error;
+        try {
+          await createDefaultEnvironment(`${__dirname}/../../../tests/stub-cards/card-definition-id-does-not-match-package-name`, factory.getModels());
+        } catch (e) {
+          error = e;
+        }
+
+        expect(error.message).to.match(/that is not scoped for this source::package/);
+      });
+
+      it('throws when there are mulitple card definitions in schema file', async function () {
```

Comment:
This error becomes impossible if we switch to returning a json:api document from schema.js. Or rather, the error condition would be slightly different (it would just need to check that `data` is singular and has type `card-definitions`).